.github/workflows/cookiecutter-infra-test.yml

```diff
@@ -0,0 +1,57 @@
+name: cookiecutter-deploy
+
+on:
+  workflow_call:
+    inputs:
+      APP_NAME:
+        required: true
+        type: string
+      SERVICE:
+        required: true
+        type: string
+      RUNNER_LABEL:
+        required: true
+        type: string
+      ENVIRONMENT:
+        required: false
+        type: string
+        default: prod
+      USE_OIDC:
+        description: 'Use OIDC for AWS authentication'
+        required: false
+        type: boolean
+        default: false
+    secrets:
+      AWS_GITHUBRUNNER_PAT:
+        required: true
+      AWS_GITHUBRUNNER_PAT_USER:
+        required: true
+      AWS_REGION:
+        required: true
+      AWS_ACCOUNT_ID:
+        required: true
+
+jobs:
+  infra-test:
+    name: infra-test
+    runs-on: ${{ inputs.RUNNER_LABEL }}
+
+    steps:
+    - name: AWS Authentication
+      uses: ./.github/actions/authenticate_aws
+      with:
+        USE_OIDC: ${{ inputs.USE_OIDC }}
+        ENVIRONMENT: ${{ inputs.ENVIRONMENT }}
+        AWS_REGION: ${{ secrets.AWS_REGION }}
+        AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+        AWS_GITHUBRUNNER_USER_ACCESS_KEY: ${{ secrets.AWS_GITHUBRUNNER_USER_ACCESS_KEY }}
+        AWS_GITHUBRUNNER_USER_SECRET_ID: ${{ secrets.AWS_GITHUBRUNNER_USER_SECRET_ID }}
+
+    - name: Check endpoint response
+      id: health_check
+      run: aws ecs wait services-stable --cluster i-dot-ai-dev-ecs-cluster --services i-dot-ai-dev-${{inputs.APP_NAME}}-${{inputs.SERVICE}}-ecs-service
+
+    - name: Notify on failure
```

Comment:
I'm not sure this step would do anything if the endpoint response fails. Do we need it? I'm not sure where it would get `$response` from either

Reply:
That is a good point, thank you, that's back when I was using `curl` still :(